receiver/receivercreator/factory.go

```diff
@@ -72,5 +74,36 @@ func createMetricsReceiver(
 	cfg component.Config,
 	consumer consumer.Metrics,
 ) (receiver.Metrics, error) {
-	return newReceiverCreator(params, cfg.(*Config), consumer)
+	var err error
+	var recv receiver.Logs
+	rCfg := cfg.(*Config)
+	r := receivers.GetOrAdd(cfg, func() component.Component {
```

Comment:
If the receiver creator instance is in both metrics and logs pipelines I believe only one will be created since the `cfg` is the same underlying map, making the other create call a noop. Instead I think we need to mutate the instance by associating the next consumer.